feat(content): add asset definitions (#156)
glpi-project/doc: source/modules/configuration/asset_definitions/asset_definitions.rst

  In the legacy API, the field name is prefixed by ``custom_``; to avoid conflicts with standard fields.
* **Type**: string, date, URL, dropdown, yes/no, text, date and time, number. Cannot be modified once saved
* **Full width**: authorises the field to be extended along the entire length of the form
* **Mandatory**: make it mandatory or not to fill in the field before saving the equipment file
* **Readonly for these profiles**: select one or more profiles with read-only access to this field.
  The permissions in the profiles tab take precedence over this field.
* **Hidden for these profiles**: select one or more profiles whose fields will be hidden
  The authorisations in the profiles tab take precedence over this field. It will therefore be visible to a profile even if it is selected in this field.
* **Default values**: specify a default value

.. note:: For dropdown list, you need to select the item in the list. You can specify if multiple selection is authorised. You can select a default value if necessary

   .. image:: images/dropdown_field.png
      :alt: Dropdown list selection
      :scale: 61%

>    .. collapse:: Dropdown list details

Reviewer: I assume this was directed to be exhaustive, but I am making another comment here anyways that this will be annoying to maintain and makes it annoying to read the documentation when such a long list is always visible.

Reviewer: It also doesn't mention the fast you can create custom dropdown types and those will be available as selections here.

       * Editions
       * Kernels
       * Kernels versions
       * Update sources
       * Networking interfaces
       * Networks
       * Network port types
       * VLANs
       * Line operators
       * Domain types
       * Domain relations
       * Records types
       * Fiber types
       * Cables types
       * Cable strands
       * Socket models
       * IP networks
       * Internet domains
       * Wifi networks
       * Networks names
       * Software categories
       * Users titles
       * User categories
       * LDAP criteria
       * Ignored values for the unicity
       * Fields storage of the login in the HTTP request
       * Plugs
       * Appliance types
       * Appliance environments
       * Resolutions
       * Image formats
       * USB vendors
       * PCI vendors
       * Webhook categories

Delete a custom field
~~~~~~~~~~~~~~~~~~~~~

.. warning:: It is **not possible** to **delete** a field **created by default**. Only fields added by the user can be deleted. However, it is possible to hide any field.

* To delete a custom field, click on the hide icon

.. image:: images/hide_field.png
   :alt: Hide a custom field
   :scale: 100 %

> * Then, click on the bin. Note that this action is irreversible

Reviewer: ```suggestion
* Then, click on the trashbin icon. Note that this action is irreversible
```

.. image:: images/delete_field.png
   :alt: Delete a custom field
   :scale: 100 %

Hide or show a field
~~~~~~~~~~~~~~~~~~~~

> Each field can be hidden in the equipment file.

Reviewer: ```suggestion
Each field can be hidden in the asset form.
```

* To hide a field, click on the hide icon

.. image:: images/hide_field.png
   :alt: Hide a custom field
   :scale: 100 %

* To restore an hidden field, drag and drop this field in the list

Modify a field
~~~~~~~~~~~~~~

You can change all the fields, but some information cannot be changed in a default field.

In a default field, you can modify :

* **Full width**
* **Mandatory**
* **Readonly for these profiles**
* **Hidden for these profiles**

.. image:: images/modify_default_field.png
   :alt: Modify a default field
   :scale: 60 %

In a custom field, you can modify :

* **Label**

> * **System name** (will be modify by the label)

Reviewer: ```suggestion
* **System name** (will be modified automatically when changing the label)
```

* **Full width**
* **Mandatory**
* **Readonly for these profiles**
* **Hidden for these profiles**
* **Default value**

.. image:: images/modify_custom_field.png
   :alt: Modify a default field
   :scale: 60 %

.. warning:: The **type** of field **cannot be modified** once it has been saved

Change the order
~~~~~~~~~~~~~~~~~

To change the order of the list of fields, drag and drop your field to the desired position.

.. image:: images/change_order.gif
   :alt: Change order field list
   :scale: 38 %

Profiles
--------

> The profiles tab is used to authorise or disable certain actions on the asset created

Reviewer: ```suggestion
The profiles tab is used to authorise certain permissions on the assets of this type
```

.. image:: images/profiles.png
   :alt: Update profiles
   :scale: 37 %

You can define the following permissions for each profile:

* Create
* View all
* Update all
* Purge
* Delete
* View assigned
* View owned
* Update assigned
* Update owned

> You can add **profiles that can associate Server with tickets, problems or changes**. This tab allow multiple selection

Reviewer: ```suggestion
You can add **profiles that can associate Server with tickets, problems or changes**. 
```
The ability to have multiple selections is implied by the label "Profiles [...]"

Author: When you speak English fluently, it's clear, but not everyone understands it as well as native speakers. I prefer to leave this information out.

Translations
------------

> You can translate the **asset name** and the **system name**

Reviewer: You cannot translate the internal name of an asset

Author: In "translations" tab, it says **asset name** but you have to understand **label**

Author: Modified
